sklearn/linear_model/logistic.py

```diff
@@ -1538,6 +1538,15 @@ def fit(self, X, y, sample_weight=None):
             raise ValueError("class_weight provided should be a "
                              "dict or 'balanced'")
 
+        # compute the class weights for the entire dataset y
+        if self.class_weight in ("auto", "balanced"):
```

Comment:
Yes, but in ` logistic_regression_path` it would be computed on the folds, and not on the entire dataset, which is the cause of the "bug" #5450, and the cause of the line in common tests that excludes LogisticRegressionCV from the class_weight tests.